trigger/patch.go

```diff
@@ -118,6 +118,47 @@ func (t *patchTriggers) patchOutcome(sub *event.Subscription) (*notification.Not
 		}
 	}
 
+	if t.patch.IsParent() || t.patch.IsChild() {
+		// check that the parent and all children are done. only the last patch to finish should send the notification
+		var childrenOrSiblings []string
+		var collectiveStatus string
+		var parentPatch patch.Patch
+		if t.patch.IsParent() {
+			childrenOrSiblings = t.patch.Triggers.ChildPatches
+		} else {
+			parentPatchId := t.patch.Triggers.ParentPatch
+			parentPatch, err := patch.FindOneId(parentPatchId)
+			if err != nil {
+				return nil, errors.Wrap(err, "can't get parent patch")
+			}
+			if parentPatch == nil {
+				return nil, errors.Errorf(fmt.Sprintf("parent patch '%s' does not exist", parentPatchId))
+			}
+			if parentPatch.Status != evergreen.PatchSucceeded && parentPatch.Status != evergreen.PatchFailed {
```

Comment:
I just merged [a thing](https://github.com/evergreen-ci/evergreen/blob/8c1f4d8771d2eca68e589f196aa24cfc9425538d/globals.go#L313) this could use